ls1intum/Artemis
# `Assessment`: Fix error message during programming assessment

### Checklist
#### General


#### Client

### Motivation and Context


There was a bug, when viewing the result details of a programming exercise while assessment. (See Testing steps)
Fixing this bug lead to another bug where the dates of the submission could not get converted to the server format. Therefore I refactored and unified the conversion code to make further bugs in these methods less likely.

### Description

This PR unifies all conversion methods in the client.
- All conversion methods follow a consistent naming scheme
- All date conversion is done inside a util class to simplify the refactored methods and reduce duplications
Additionally it fixes a bug during the assessment of a programming exercise.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1Student
- 1 Programming Exercise with manual assessment

1. Log in to Artemis
2. Create the exercise
3. Participate as a student
4. Assess the submission
5. Klick on the result string during the assessment to see *no* error message

### Review Progress





#### Code Review
#### Manual Tests
